### PR TITLE
Update to DiagnosticSource 5.0 preview4

### PR DESCRIPTION
**Pull request by cijothomas**
As requested in https://github.com/open-telemetry/opentelemetry-dotnet/pull/660, I am updating the newest preview version of DiagnosticSource. This package is not yet available in nuget.org, so added a new nuget source (temporary, until nuget.org has this package).

This PR only updates DiagnosticSource version. Actual code leveraging the new Activity, ActivitySource (similar to the preview PR 660), is going to follow in separate PRs.

Fixes # .

## Changes
(Please provide a brief description of the changes here.)

### Checklist
- [ ] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public API reviewed

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
This change sets the repo in a very strong dependency on .NET release dates. And make it much harder for early adopters to try.

Can we keep this change in a separate PR for now? I want to keep the repo in a working state.

**Comment by bruno-garcia** [comment]
@SergeyKanzhelev I believe this is a requirement for us to move from `OTel.Api.Span` to `Activity`

**Comment by reyang** [comment]
> This change sets the repo in a very strong dependency on .NET release dates. And make it much harder for early adopters to try.

> Can we keep this change in a separate PR for now? I want to keep the repo in a working state.

I think it is fine to have a dependency to .NET preview while OpenTelemetry .NET SDK is at alpha/beta stage?
For people who already use the current (alpha0.2) SDK, they can still use that.
We are making change to the next alpha release which will take breaking changes, and this requires a dependency on .NET preview.

File: NuGet.config [hunk 1]
```diff
@@ -1,8 +1,9 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
     <clear />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
+    <add key="Dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
```
Comment (SergeyKanzhelev) [inline NuGet.config:6]: next week it will be on NuGet. Should be good to merge now than

**Pull request merged**